index.js

```diff
@@ -12,3 +15,24 @@ const pagination = document.querySelector('[data-js="pagination"]');
 const maxPage = 1;
 const page = 1;
 const searchQuery = "";
+
+async function fetchCharacters() {
+  try {
+    const response = await fetch(ramCharacters);
+    if (response.ok) {
+      const data = await response.json();
+      const characterData = data.results;
+      cardContainer.innerHTML = "";
+      characterData.forEach((character) => {
+        const card = createCharacterCard(character);
+        cardContainer.append(card);
+      });
+      return data;
+    } else {
+      console.error("response not OK");
+    }
+  } catch (error) {
+    console.error("API not available: ", error);
+  }
+}
```

Comment:
I like that you use a try/catch block here 🙂 

In this case, you directly create the character cards from the data. You do not need to return the data, as you won't do anything else with it.
```suggestion
async function fetchCharacters() {
  try {
    const response = await fetch(ramCharacters);
    if (response.ok) {
      const data = await response.json();
      const characterData = data.results;
      cardContainer.innerHTML = "";
      characterData.forEach((character) => {
        const card = createCharacterCard(character);
        cardContainer.append(card);
      });
    } else {
      console.error("response not OK");
    }
  } catch (error) {
    console.error("API not available: ", error);
  }
}
```